core/types/calls.cc

```diff
@@ -3291,7 +3294,7 @@ class Shape_squareBracketsEq : public IntrinsicMethod {
             // people's codebases to it.
             //
             // TODO(jez) This could be another "if you're in `typed: strict` you need typed shapes"
-            res.returnType = Types::untypedUntracked();
+            res.returnType = Types::untyped(Symbols::Magic_UntypedSource_squareBracketsEq());
```

Comment:
Any class can define a method called `[]=`. So the name "UntypedSource_squareBracketsEq" is a bit too broad. This blame is _specifically_ for `[]=` on shapes. We will want the name of this blame to reflect that.